.github/workflows/lint.yml

```diff
@@ -0,0 +1,21 @@
+name: Lint
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-go@v5
+      - run: go install mvdan.cc/sh/v3/cmd/shfmt@latest
+      - run: sudo apt-get install -y shellcheck
+      - name: run shellcheck
+        run: shellcheck ./bin/*
+      - name: run shfmt
+        run: shfmt -i 2 --diff ./bin/*
```

Comment:
👍  very nice to have.   Maybe not in this PR but would be nice to have a Makefile so we can do `make lint` before we push stuff.

Reply:
Good idea! Added in the latest commit.